cmd/kubeadm/app/images/images.go

```diff
@@ -39,10 +40,11 @@ func GetCoreImage(image string, cfg *kubeadmapi.MasterConfiguration, overrideIma
 	}
 	repoPrefix := kubeadmapi.GlobalEnvParams.RepositoryPrefix
 	return map[string]string{
-		KubeEtcdImage:              fmt.Sprintf("%s/%s-%s:%s", repoPrefix, "etcd", runtime.GOARCH, etcdVersion),
+		KubeEtcdImage:              fmt.Sprintf("quay.io/coreos/etcd:%s", cfg.Etcd.Cluster.Version),
```

Comment:
Actually, etcd-operator currently hardcodes the registry URL [here](https://github.com/coreos/etcd-operator/blob/9bfc87aa1473da7931b3d1f285829472ab2099fc/pkg/util/k8sutil/k8sutil.go#L113). So before adding this option we'd need to add it to the etcd-operator codebase first. Are you ok with this being done in a follow-up PR?